src/mqt/bench/evaluation/evaluation.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import annotations
+
+import pickle
+import sys
+from typing import TYPE_CHECKING, Any
+
+if TYPE_CHECKING or sys.version_info >= (3, 10, 0):  # pragma: no cover
+    pass
+else:
+    pass
+
+from pathlib import Path
+
+import numpy as np
+from joblib import Parallel, delayed
+from mqt.bench import utils
+from qiskit import QuantumCircuit, QuantumRegister
+from qiskit.transpiler.passes import RemoveBarriers
+
+if TYPE_CHECKING or sys.version_info >= (3, 10, 0):  # pragma: no cover
+    from importlib import resources
+else:
+    import importlib_resources as resources
+
+
+def create_statistics() -> None:
+    source_circuits_list = [file for file in Path(utils.get_qasm_output_path()).iterdir() if file.suffix == ".qasm"]
+
+    res = Parallel(n_jobs=-1, verbose=100)(delayed(evaluate_qasm_file)(filename) for filename in source_circuits_list)
+    target_dir = Path(resources.files("mqt.bench") / "evaluation/")
+    with Path(target_dir / "evaluation_data.pkl").open("wb") as f:
+        pickle.dump(res, f)
+
+
+def evaluate_qasm_file(filename: str) -> tuple[str, int, int, int, int, float, float, float, float, float]:
+    print(filename)
+    qc = QuantumCircuit.from_qasm_file(filename)
+    qc.remove_final_measurements(inplace=True)
+    (program_communication, critical_depth, entanglement_ratio, parallelism, liveness) = calc_supermarq_features(qc)
+    return (
+        filename,
+        int(str(filename).split("_")[-1].split(".")[0]),
+        qc.depth(),
+        sum(qc.count_ops().values()),
+        qc.num_nonlocal_gates(),
+        program_communication,
+        critical_depth,
+        entanglement_ratio,
+        parallelism,
+        liveness,
+    )
+
+
+def calc_qubit_index(qargs: list[Any], qregs: list[QuantumRegister], index: int) -> Any:
+    offset = 0
+    for reg in qregs:
+        if qargs[index] not in reg:
+            offset += reg.size
+        else:
+            qubit_index = offset + reg.index(qargs[index])
+            return qubit_index
+    error_msg = "Qubit not found."
```

Comment:
That error message could be a little more descriptive. As it is, it contains almost no information about the underlying problem.

Reply:
Improved on that.